networkit/cpp/planarity/LeftRightPlanarityCheck.cpp

```diff
@@ -0,0 +1,262 @@
+/*  LeftRightPlanarityCheck.cpp
+ *
+ *	Created on: 03.01.2025
+ *  Authors: Andreas Scharf (andreas.b.scharf@gmail.com)
+ *
+ */
+
+#include <networkit/planarity/LeftRightPlanarityCheck.hpp>
+
+namespace NetworKit {
+
+void LeftRightPlanarityCheck::run() {
+    // Euler-criterion
+    if (graph_->numberOfNodes() > 2 && graph_->numberOfEdges() > 3 * graph_->numberOfNodes() - 6) {
+        hasRun = true;
+        isPlanar_ = false;
+        return;
+    }
+
+    heights.assign(graph_->upperNodeIdBound(), noneHeight);
+    graph_->forNodes([&](const node currentNode) {
```

Comment:
Avoid using `const` for primitive types in functions input parameters, also below.